plugins/UpdateChecker/UpdateCheckerJob.py

```diff
@@ -96,4 +83,27 @@ def run(self):
             no_new_version = False  # Just to suppress the message below.
 
         if no_new_version and not self.silent:
-            Message(i18n_catalog.i18nc("@info", "No new version was found."), title = i18n_catalog.i18nc("@info:title", "Version Upgrade")).show()
\ No newline at end of file
+            Message(i18n_catalog.i18nc("@info", "No new version was found."), title = i18n_catalog.i18nc("@info:title", "Version Upgrade")).show()
+
+    def showUpdate(self, newest_version: Version, download_url: str) -> None:
+        preferences = Application.getInstance().getPreferences()
+        latest_version_shown = preferences.getValue("info/latest_update_version_shown")
+        if latest_version_shown == newest_version and not self.display_same_version:
+            return #Don't show this update again. The user already clicked it away and doesn't want it again.
+        preferences.setValue("info/latest_update_version_shown", str(newest_version))
```

Comment:
I would suggest to move these 5 lines out to a different method, in order to separate concerns and being sure that the `showUpdate` only shows the update message.